OSDOCS-13050:Packet translation (#86910)
openshift/openshift-docs: modules/network-observability-packet-translation-overview.adoc

// Module included in the following assemblies:
//
// network_observability/observing-network-traffic.adoc

:_mod-docs-content-type: CONCEPT
[id="network-observability-packet-translation-overview_{context}"]
= Endpoint translation (xlat)

> You can gain visibility into the endpoints serving traffic in a consolidated view using Network Observability and eBPF (extended Berkeley Packet Filter). Typically, when traffic flows through a service, egressIP, or load balancer, the traffic flow information is abstracted as it is routed to one of the available pods; if you try to get information about the traffic, you can only view service related info, such as service IP and port, and not information about the specific pod that is serving the request. Often the information for both the service traffic and the virtual service endpoint is captured as two separate flows, which complicates troubleshooting. 

Reviewer: ```suggestion
You can gain visibility into the endpoints serving traffic in a consolidated view using Network Observability and extended Berkeley Packet Filter (eBPF). Typically, when traffic flows through a service, egressIP, or load balancer, the traffic flow information is abstracted as it is routed to one of the available pods. If you try to get information about the traffic, you can only view service related info, such as service IP and port, and not information about the specific pod that is serving the request. Often the information for both the service traffic and the virtual service endpoint is captured as two separate flows, which complicates troubleshooting. 
```

To solve this, endpoint xlat can help in the following ways:

- Capture the network flows at the kernel level, which has a minimal impact on performance.
- Enrich the network flows with translated endpoint information, showing not only the service but also the specific backend pod, so you can see which pod served a request.

> As network packets are processed, the eBPF hook enriches flow logs with metadata about the translated endpoint that includes the following pieces of information that can easily be viewed in the *Network Traffic* page in a single row:

Reviewer: ```suggestion
As network packets are processed, the eBPF hook enriches flow logs with metadata about the translated endpoint that includes the following pieces of information that you can view in the *Network Traffic* page in a single row:
```

- active voice

// Module included in the following assemblies:
//
// network_observability/observing-network-traffic.adoc

:_mod-docs-content-type: CONCEPT
[id="network-observability-packet-translation-overview_{context}"]
= Endpoint translation (xlat)

> You can gain visibility into the endpoints serving traffic in a consolidated view using Network Observability and eBPF (extended Berkeley Packet Filter) to enrich Kubernetes service flows. 

Reviewer 2: Can remove `to enrich Kubernetes service flows`

// Module included in the following assemblies:
//
// network_observability/observing-network-traffic.adoc

:_mod-docs-content-type: CONCEPT
[id="network-observability-packet-translation-overview_{context}"]
= Endpoint translation (xlat)

> You can gain visibility into the endpoints serving traffic in a consolidated view using Network Observability and eBPF (extended Berkeley Packet Filter) to enrich Kubernetes service flows. Typically, when traffic flows through a Kubernetes Service, a kube-proxy or other load balancers abstracts the traffic flow information. For example, a request to `my-service` on `ClusterIP` is transparently routed to one of the available pods. However, often the information for both the service traffic and the virtual service endpoint is captured as two separate flows, which complicates troubleshooting. 

Reviewer 2: Typically when traffic flows through a Service, EgressIP or Load balancer then the traffic is abstracted and routed to one of the availabel pods.

Reviewer 2: However it does not provide granular info about which actual pod is serving that request

Reviewer 2: We just service related info( service IP, port) and not the pod info

To solve this, endpoint xlat can help in the following ways:

- Capture the network flows at the kernel level, which has a minimal impact on performance since.

- Enrich the network flows with translated endpoint information, showing not only the service but also the specific backend pod, so you can see which pod served a request.

As network packets are processed, the eBPF hook enriches flow logs with metadata about the translated endpoint that includes the following pieces of information that can easily be viewed in the *Network Traffic* page in a single row:

- Source Pod IP

- Source Port

- Destination Pod IP

- Destination Port

- link:https://lwn.net/Articles/370152/#:~:text=A%20zone%20is%20simply%20a,to%20seperate%20conntrack%20defragmentation%20queues.[Conntrack Zone ID]

> - ICMP code

Reviewer 2: this is removed
